Runtime/Model/Database/BacktraceDatabaseAttachmentManager.cs

```diff
@@ -93,23 +95,60 @@ private string GetScreenshotPath(string dataPrefix)
                 }
                 else
                 {
-                    // Create a texture the size of the screen, RGB24 format
-                    int width = Screen.width;
-                    int height = Screen.height;
-                    Texture2D tex = new Texture2D(width, height, TextureFormat.RGB24, false);
-
-                    // Read screen contents into the texture
-                    tex.ReadPixels(new Rect(0, 0, width, height), 0, 0);
-                    tex.Apply();
-
-                    // Encode texture into JPG and save it on hard drive
-                    var bytes = tex.EncodeToJPG();
-                    using (var fs = new FileStream(screenshotPath, FileMode.Create, FileAccess.Write))
+                    Texture2D result;
+                    if (ScreenshotMaxHeight < Screen.height)
                     {
-                        fs.Write(bytes, 0, bytes.Length);
+
+                        float ratio = Screen.width / Screen.height;
+                        int targetHeight = Mathf.Min(Screen.height, ScreenshotMaxHeight);
+                        int targetWidth = Mathf.RoundToInt(targetHeight * ratio);
+
+#if UNITY_2019_1_OR_NEWER
+                        RenderTexture screenTexture = RenderTexture.GetTemporary(Screen.width, Screen.height);
+                        ScreenCapture.CaptureScreenshotIntoRenderTexture(screenTexture);
```

Comment:
Is it worth doing this asynchronously? https://docs.unity3d.com/ScriptReference/ScreenCapture.CaptureScreenshotIntoRenderTexture.html

Especially if it is a non-fatal exception, maybe we don't want to block important game threads. I imagine the downsampling of the image will add some extra processing time as well. It might be worth profiling to see if we're actually saving time with this during runtime.

Reply:
The problem that I have with async in this solution is that we will need to write async callbacks to continue error submission. Overall our solution uses coroutines. Just after this database operation, we will yield a return from the Backtrace client to free the current frame.

Reply:
Do we save time here? It looks like we still capture the screenshot at full resolution and then take additional time to do downsampling. It looks like we maybe are adding time overall. Maybe we don't care but I'm only asking because of this user story:

"As a user, I would like to define attachments settings that allow the game to save resources/time when the game generates image attachment."

I understand we would save resources by saving downsampled images, but are we saving time? I think we are probably adding time

Reply:
I guess we can save time because we are spending less time on the file write. But it might be worth doing a simple test to check if we are actually saving time overall including the downsampling algorithm. Otherwise I'm sure the functionality works and there's no issues there.

Reply:
I guess we can save time because we are spending less time on the file write. But it might be worth doing a simple test to check if we are actually saving time overall including the downsampling algorithm. Otherwise I'm sure the functionality works and there's no issues there.

Reply:
Do we save time? 
I don't think so:
(without image modifications) 
![image](https://user-images.githubusercontent.com/9386268/121222108-8788e280-c886-11eb-9f78-47f50f8897a6.png)

(without image modification)
![image](https://user-images.githubusercontent.com/9386268/121221724-1fd29780-c886-11eb-8e79-4c9215526995.png)

Do we save anything at all? Yes - transfer, attachment size, disk usage.

Reply:
Sounds good, if user enables this setting they will have smaller attachment sizes which can enable user to capture more screenshots and otherwise save on disk usage, as well as save transfer time at crash time. But user will not save during run time. Thank you for providing the numbers!